src/TestFramework/TestFramework/Attributes/TestMethod/OSConditionAttribute.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// This attribute is used to ignore a test class or a test method, with an optional message.
+/// </summary>
+/// <remarks>
+/// This attribute isn't inherited. Applying it to a base class will not cause derived classes to be ignored.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = false)]
+public sealed class OSConditionAttribute : ConditionBaseAttribute
+{
+    private readonly OperatingSystems _operatingSystems;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OSConditionAttribute"/> class.
+    /// </summary>
+    /// <param name="operatingSystems">The operating systems that this test supports.</param>
+    public OSConditionAttribute(OperatingSystems operatingSystems, Mode mode)
```

Comment:
I was thinking to have mode first for consistency with other attributes we will/would ship and will be string based (params).